source/game_sa/Game.cpp

```diff
@@ -74,38 +67,100 @@ void CGame::InjectHooks() {
     RH_ScopedInstall(CanSeeWaterFromCurrArea, 0x53C4B0);
     RH_ScopedInstall(Init1, 0x5BF840);
     RH_ScopedInstall(Init2, 0x5BA1A0);
-    RH_ScopedInstall(Init3, 0x5BA400, { .reversed = false });
+    RH_ScopedInstall(Init3, 0x5BA400);
     RH_ScopedInstall(Initialise, 0x53BC80);
-    RH_ScopedInstall(InitialiseCoreDataAfterRW, 0x5BFA90, { .reversed = false });
-    RH_ScopedInstall(InitialiseEssentialsAfterRW, 0x5BA160, { .reversed = false });
+    RH_ScopedInstall(InitialiseCoreDataAfterRW, 0x5BFA90);
+    RH_ScopedInstall(InitialiseEssentialsAfterRW, 0x5BA160);
     RH_ScopedInstall(InitialiseOnceBeforeRW, 0x53BB50);
-    RH_ScopedInstall(InitialiseRenderWare, 0x5BD600, { .reversed = false });
+    RH_ScopedInstall(InitialiseRenderWare, 0x5BD600);
     RH_ScopedInstall(InitialiseWhenRestarting, 0x53C680);
-    RH_ScopedInstall(Process, 0x53BEE0, { .reversed = false });
-    RH_ScopedInstall(ReInitGameObjectVariables, 0x53BCF0, { .reversed = false });
+    RH_ScopedInstall(Process, 0x53BEE0);
+    RH_ScopedInstall(ReInitGameObjectVariables, 0x53BCF0);
     RH_ScopedInstall(ReloadIPLs, 0x53BED0);
-    RH_ScopedInstall(ShutDownForRestart, 0x53C550, { .reversed = false });
-    RH_ScopedInstall(Shutdown, 0x53C900, { .reversed = false });
+    RH_ScopedInstall(ShutDownForRestart, 0x53C550);
+    RH_ScopedInstall(Shutdown, 0x53C900);
     RH_ScopedInstall(ShutdownRenderWare, 0x53BB80);
     RH_ScopedInstall(DrasticTidyUpMemory, 0x53C810);
     RH_ScopedInstall(FinalShutdown, 0x53BC30);
     RH_ScopedInstall(TidyUpMemory, 0x53C500);
+
+    RH_ScopedGlobalInstall(CameraDestroy, 0x72FD90);
+    RH_ScopedGlobalInstall(ValidateVersion, 0x5BA0BC);
 }
 
 // 0x72FD90
-static void CameraDestroy(RwCamera *pRwCamera) {
-    plugin::Call<0x72FD90, RwCamera *>(pRwCamera);
+static void CameraDestroy(RwCamera *rwCamera) {
+    if (!rwCamera)
+        return;
+
+    if (auto obj = rwCamera->object.object; obj.parent) {
+        rwObjectHasFrameSetFrame(&obj, NULL);
+        RwFrameDestroy((RwFrame*)obj.parent);
+    }
+
+    if (auto frameBuffer = rwCamera->frameBuffer; frameBuffer) {
```

Comment:
Macro for `frameBuffer`?
Also, no need for the `frameBuffer` at the end.
```suggestion
    if (auto frameBuffer = rwCamera->frameBuffer) {
```